Mathlib/Algebra/Quaternion.lean

```diff
@@ -639,55 +654,63 @@ theorem im_star : (star a).im = -a.im :=
   QuaternionAlgebra.ext neg_zero.symm rfl rfl rfl
 
 @[simp]
-theorem star_mk (a₁ a₂ a₃ a₄ : R) : star (mk a₁ a₂ a₃ a₄ : ℍ[R,c₁,c₂]) = ⟨a₁, -a₂, -a₃, -a₄⟩ :=
-  rfl
+theorem star_mk (a₁ a₂ a₃ a₄ : R) : star (mk a₁ a₂ a₃ a₄ : ℍ[R,c₁,c₂,c₃]) =
+    ⟨a₁ + c₂ * a₂, -a₂, -a₃, -a₄⟩ := rfl
 
-instance instStarRing : StarRing ℍ[R,c₁,c₂] where
+instance instStarRing : StarRing ℍ[R,c₁,c₂,c₃] where
   star_involutive x := by simp [Star.star]
-  star_add a b := by ext <;> simp [add_comm]
+  star_add a b := by ext <;> simp [add_comm] ; ring
   star_mul a b := by ext <;> simp <;> ring
 
-theorem self_add_star' : a + star a = ↑(2 * a.re) := by ext <;> simp [two_mul]
+theorem self_add_star' : a + star a = ↑(2 * a.re + c₂ * a.imI) := by ext <;> simp [two_mul]; ring
 
-theorem self_add_star : a + star a = 2 * a.re := by simp only [self_add_star', two_mul, coe_add]
+theorem self_add_star : a + star a = 2 * a.re + c₂ * a.imI := by
+  simp [self_add_star']; rfl
 
-theorem star_add_self' : star a + a = ↑(2 * a.re) := by rw [add_comm, self_add_star']
+theorem star_add_self' : star a + a = ↑(2 * a.re + c₂ * a.imI) := by rw [add_comm, self_add_star']
 
-theorem star_add_self : star a + a = 2 * a.re := by rw [add_comm, self_add_star]
+theorem star_add_self : star a + a = 2 * a.re + c₂ * a.imI := by rw [add_comm, self_add_star]
 
-theorem star_eq_two_re_sub : star a = ↑(2 * a.re) - a :=
+theorem star_eq_two_re_sub : star a = ↑(2 * a.re + c₂ * a.imI) - a :=
   eq_sub_iff_add_eq.2 a.star_add_self'
 
 instance : IsStarNormal a :=
   ⟨by
-    rw [a.star_eq_two_re_sub]
-    exact (coe_commute (2 * a.re) a).sub_left (Commute.refl a)⟩
+    rw [commute_iff_eq, a.star_eq_two_re_sub]
+    ext <;> simp <;> ring⟩
```

Comment:
My `ofNat` suggestion above breaks this proof, and you should fix it by making ofNat lemmas just like the natCast lemmas in line 400-440 or so:
```
@[simp]
theorem ofNat_re (n : ℕ) [n.AtLeastTwo] : (ofNat(n) : ℍ[R,c₁,c₂,c₃]).re = ofNat(n) := rfl

@[simp]
theorem ofNat_imI (n : ℕ) [n.AtLeastTwo] : (ofNat(n) : ℍ[R,c₁,c₂,c₃]).imI = 0 := rfl

@[simp]
theorem ofNat_imJ (n : ℕ) [n.AtLeastTwo] : (ofNat(n) : ℍ[R,c₁,c₂,c₃]).imJ = 0 := rfl

@[simp]
theorem ofNat_imK (n : ℕ) [n.AtLeastTwo] : (ofNat(n) : ℍ[R,c₁,c₂,c₃]).imK = 0 := rfl
```
etc